---
title: "Chignik 2018 sockeye baseline update"
date: "08/24/2018"
editor_options:
  chunk_output_type: inline
output:
  html_document:
    df_print: paged
subtitle: Sockeye_UW_7SNPs
creator: Chase Jalbert
---

### Introduction ###

This script is an update to the 2012 Chignik baseline. I am including 7 new SNPs from Wes Larson's RAD work.    

Generally, here are some helpful notes and the major steps invloved in this analysis.    

old script but good to follow:    
    
    V:\Analysis\4_Westward\Sockeye\Chignik Inseason 2012-2018\Baseline

fairly recent-ish for more updated code:    
    
    V:\Analysis\4_Westward\Sockeye\KMA Commercial Harvest 2014-2016\GitHub-KMA-Sockeye-Mixtures


* get names of new markers, Locus control    
  +  create with sock 2011 96
   - CreateLocusControl.GCL(markersuite = "Sockeye2011_96SNPs", username = username, password = password) 
  +  extract locus names as object    
  +  rm(locuscontrol)    
  +  create locus control with 7 new ones    
   - CreateLocusControl.GCL(markersuite = "Sockeye2011_96SNPs", username = username, password = password)    
  + extract locus names as object    
  + rm(locuscontrol)    
  + make new list of locus names using correct spelling from 96 andf 7 lists    


* LOKI2R Makes .gcl objects and sillyvec will be list of sillys    
-(can get list from :readproject")    

   
   Key steps in script: 
   QA
   Analysis
     - HWE
     - Pool
     - HWE
     - LD
     - Final Markers
    

```{r setup}
#Uncomment to delete everything from environment...
#ls()
#rm(list=ls(all=TRUE))
#search()

if (!require("pacman"))
  install.packages("pacman")
library(pacman) # install pacman, if not installed
p_load(adegenet, tidyverse, ape, genepop, pegas, reshape) # use pacman to load or install+load necessary packages

#This sources all of the new GCL functions to this workspace
source("C:/Users/csjalbert/Documents/r_proj/function_gcl.r")
```


LOKI username and password:    
<i>intentionally hidden from output</i>
```{r user_pwd, include = FALSE}
# yourloki username 
username <- "csjalbert"

# your loki password
.password <- "esox.1234"
```



# Full Marker suite using <b>Sockeye2011_96SNPs</b> and <b>Sockeye_UW_7SNPs</b>. 
```{r marker suite}
# create locus control for Sockeye2011_96SNPs.
CreateLocusControl.GCL(markersuite = "Sockeye2011_96SNPs",
                       username = username,
                       password = .password)

# get locus names
loci_sock2011_96 <- LocusControl$locusnames

# remove object
rm(LocusControl)

# create locus control for Sockeye_UW_7SNPs to get locus names then remove object.
CreateLocusControl.GCL(markersuite = "Sockeye_UW_7SNPs",
                       username = username,
                       password = .password)

loci_sockuw_7 <- LocusControl$locusnames

rm(LocusControl)

# combine into a single file
Sockeye_all_SNPs <- c(loci_sock2011_96, loci_sockuw_7)

# create locus control for all sockeye markers
CreateLocusControl.GCL(locusnames = Sockeye_all_SNPs,
                       username = username,
                       password = .password)

# clean up old locus names, using ls() which allows Regex
rm(list = ls(pattern = "^loci_sock"))
```

```{r}
# Vector of silly names for ReadLoki function
ChignikCollections <-
      c(
        "SHAT96",
        "SHAT97E",
        "SALEC97",
        "SBOUL97",
        "SBROAD97",
        "SBSPR97",
        "SCHIA97E",
        "SCHIA97M",
        "SCHIG98",
        "SCLARK96",
        "SCLRK97E",
        "SFAN97",
        "SCHIA08",
        "SCHIG08",
        "SCLARK08",
        "SCUCU08",
        "SHAT08E",
        "SWESTF08"
      ) # note - no SWESTF97 due to lack of metadata, as noted in Tyler's 2012 baseline update.


## Pull all data for each silly code and create .gcl objects for each
LOKI2R.GCL(sillyvec = ChignikCollections,
           username = username,
           password = .password)
objects(pattern = "\\.gcl")

# remove password from file
rm(.password)
```

```{r N_by_location}
## Get sample size by locus
OriginalSampleSizebyLocus <-
  SampSizeByLocus.GCL(sillyvec = ChignikCollections, loci = Sockeye_all_SNPs)

min(OriginalSampleSizebyLocus) ## smallest is 69, fine.

write.table(
  OriginalSampleSizebyLocus,
  "../2018/Output/OriginalSampleSizebyLocus.txt",
  sep = '\t',
  col.names = NA,
  row.names = T
)

# view table of sizes by location
apply(OriginalSampleSizebyLocus, 1, range)
```

```{r}
# saving objects
save_objects("Sockeye_all_SNPs", path = "../2018/Objects")

# load previously saved objects
#load_objects(path = "../2018/Objects")

# save sillys, no quotes so each as an individual file:
save_sillys(sillyvec = ChignikCollections, path = "../2018/sillys")

# export table of sample size by locus, csv since nice tidy df already and not R dependent.
write_csv(OriginalSampleSizebyLocus, path = "../2018/Output/samplesizebylocus.csv")
```


# QA section, obtain number of individuals in collections

```{r original_N}
# get number of individuals in SILLY, for all sillys, prior to removing missing loci individuals
OriginalColSize <-
  sapply(paste(ChignikCollections, ".gcl", sep = ''), function(x)
    get(x)$n)

# export table of individuals in each SILLY
write.table(
  OriginalColSize,
  file = "../2018/Output/OriginalColSize.csv",
  sep = ",",
  col.names = NA,
  row.names = T
)
```


```{r remove_20per}
# Remove individuals with >20% missing data
MissLoci <-
  RemoveIndMissLoci.GCL(sillyvec = ChignikCollections, proportion = 0.8)

 write.table(
  lapply( MissLoci, 'length<-', max(lengths(MissLoci)))[], # assigning same size column to all missing so I can output table. represented as NAs
   file = "../2018/Output/missing.csv",
   sep = ",",
   col.names = NA,
   row.names = T
 )
```

Get the number of individuals per silly after removing the missing loci inds

```{r MissingLoci}

# post removal individuals 
ColSizePostMissLoci <-
  sapply(paste(ChignikCollections, ".gcl", sep = ''), function(x)
    get(x)$n)

# output table
write.table(
  OriginalColSize,
  file = "../2018/Output/ColSizePostMissLoci.csv",
  sep = ",",
  col.names = NA,
  row.names = T
)

# calculate difference between pre- and post- removal individuals (i.e., number dropped)
OriginalColSize - ColSizePostMissLoci
```


Create matrix for collection information

```{r CollectionInfo}
#formatting into matrix
Chignik2018SockeyeSampleSizes <-
  matrix(
    data = NA,
    nrow = 18,
    ncol = 4,
    dimnames = list(
      names(OriginalColSize),
      c("Initial", "Missing", "Duplicate", "Final")
    )
  )

Chignik2018SockeyeSampleSizes[, 1] <- OriginalColSize

Chignik2018SockeyeSampleSizes[, 2] <-
  OriginalColSize - ColSizePostMissLoci
```


Check within collections for duplicate individuals, at 95% level

```{r DupCheck}

# duplicate checks
DuplicateCheck95MinProportion <-
  CheckDupWithinSilly.GCL(
    sillyvec = ChignikCollections,
    loci = Sockeye_all_SNPs,
    quantile = NULL,
    minproportion = 0.95
  )

#str(DuplicateCheck95MinProportion)

# create a report summary of duplciates
DuplicateCheckReportSummary <-
  sapply(ChignikCollections, function(x)
    DuplicateCheck95MinProportion[[x]]$report)

## Remove duplicate individuals
RemovedDups <- RemoveDups.GCL(DuplicateCheck95MinProportion)

## Get number of individuals per silly after removing duplicate individuals
ColSizePostDuplicate <-
  sapply(paste(ChignikCollections, ".gcl", sep = ''), function(x)
    get(x)$n)

# write table containing duplicates
write.table(
  ColSizePostDuplicate,
  file = "../2018/Output/ColSizePostDuplicate.csv",
  sep = ",",
  col.names = NA,
  row.names = T
)

#check N lost
ColSizePostMissLoci - ColSizePostDuplicate

# write output to matrix created above
Chignik2018SockeyeSampleSizes[, 3] <-
  ColSizePostMissLoci - ColSizePostDuplicate

Chignik2018SockeyeSampleSizes[, 4] <- ColSizePostDuplicate
```

# Allele frequency plots

```{r allele freq setup}
#import list of silly, groups, and locations
Groups <- read_csv(file = "../2018/Objects/group_locations.csv")

#character vector of pop Names without the ".gcl" extension
sillyvec <- unlist(Groups[1])

#numeric vector of group numbers the same length as sillyvec
groupvec <- unlist(Groups[2])

#A numeric vector of R colors the same length as max(groupvec)
mycolor <- c("black", "red")

# assign colors for each pop, based on groups
PopCol = mycolor[(groupvec)]
```


```{r Allele frequency plots}
loci <- LocusControl$locusnames

Freq <- FreqPop.GCL(sillyvec = sillyvec, loci = loci)

Freq <-
  Freq[, , "Allele 1"] / (Freq[, , "Allele 1"] + Freq[, , "Allele 2"])

pdf(
  file = "../2018/freqplots/Chignik103SNPfreqplots_popnames.pdf",
  width = 11,
  height = 8.5,
  family = "Times",
  pointsize = 20
)
for (locus in loci) {
  plot(
    Freq[, locus],
    main = locus,
    col = PopCol,
    pch = 19,
    ylim = c(0, 1),
    ylab = "Freqency",
    xlab = "Collection"
  )
  text(x = Freq[, locus], labels = sillyvec, cex = 0.7, pos = 1, srt = 90, offset = 1.5, xpd = TRUE) # add population names to file
  lines(supsmu(seq(length(sillyvec)), Freq[, locus]))
}

rm(locus)

dev.off()
```

# Test for HWE

Here I am going to test for HWE using genepop package, so we dont have to export and analyze outside of the R-environement (sort of). All genepop does is export the .txt file, then read that file and run genepop, within r, but outputs to a .P file. There is no R output, so we have to read the .P file in using ReadGenepopHWE.GCL. This is handy because it avoids having to run gnepop.exe outside of the R session, hence avoids manually moving files.    

Another option is to export a genepop file, reimport as genind, then run hw.test (pegas package). A brief look shows that Pegas and genepop produce similar estimates of HWE. After talking to Kyle, it was decided that genepop is the standard so we will rely on that program / package for our analyses. 

```{r HWE}
# first convert to genepop format using GCL functions
gcl2Genepop.GCL(
  sillyvec = ChignikCollections,
  loci = loci,
  path = "../2018/genepop/ChignikCollections.txt",
  VialNums = T
)

# HW genotypic proportions using 'genepop'
test_HW("../2018/genepop/ChignikCollections.txt", which = "Proba")

# parse results using GCL function
HWE <-
  ReadGenepopHWE.GCL(file = "../2018/genepop/ChignikCollections.txt.P")

str(HWE)
```


Now we can identify any fixed loci. It looks like One_txnip_401 (18) and probably One_metA-253 (17.99451) - all others seem okay, although some fairly close to fixed.

```{r}
FixedSNPs <- sort(apply(Freq, 2, sum))

FixedSNPs <- names(FixedSNPs[FixedSNPs == 0 | FixedSNPs == 18])
```


```{r}
test_HW("../2018/genepop/ChignikCollections.txt", which = "Proba")
HWE <-
  ReadGenepopHWE.GCL(file = "../2018/genepop/ChignikCollections.txt.P")
str(HWE)

# just nuclear loci
locinuclear <- loci[LocusControl$ploidy == 2]

# convert to genepop format
gcl2Genepop.GCL(
  sillyvec = ChignikCollections,
  loci = locinuclear,
  path = "../2018/genepop/ChignikCollectionsnuclearloci.txt",
  VialNums = TRUE
)

# HW genotypic proportions using 'genepop'
test_HW("../2018/genepop/ChignikCollectionsnuclearloci.txt", which = "Proba")

# parse results using GCL function
HWE.N <-
  ReadGenepopHWE.GCL(file = "../2018/genepop/ChignikCollectionsnuclearloci.txt.P")
str(HWE.N)
```

# Linkage disequilibrium

Here I am going to test for LD among the 7 new RAD SNPs from Wes Larson's "Islands of Divergence" paper. Included in his paper, is Supplemental Table S4 which includes position information for these 7 SNPs based on their linkage map. Based on Table S4, 6 of the 7 SNPs appear to be on linkage group 13, with one on LG 12. This gives us enough information to *a priori* assume that we can only select one of the LG 13 SNPs in addition to the LG 12 SNP.

I ran Genepop from the executible for the file "genepop/ChignikCollections.txt", which has all 103 SNPS and all 18 Chignik collections. Here we can read in the results and see what is statistically in LD.

```{r read_LD}
(LD <- ReadGenepopDis.GCL(file = "../2018/genepop/ChignikCollections.txt.DIS") %>% 
  as_tibble())
```
Typically we determine that two markers are linked if they are P < 0.5 for 1/2 of all populations/collections.

```{r LD_npops}
LD <- LD %>% 
  gather(silly, pvalue, -Locus1, -Locus2, -Overall) %>% 
  group_by(Locus1, Locus2) %>% 
  mutate(n_pops_fail = sum(pvalue < 0.05)) %>% 
  spread(silly, pvalue)

LD %>% 
  ggplot(aes(x = n_pops_fail)) +
  geom_histogram() +
  geom_vline(xintercept = 9)
```

Which pairs of loci are "linked" according to the 1/2 pops P < 0.5 rule?

```{r LD_fail}
LD %>% 
  filter(n_pops_fail >= 9) %>% 
  select(Locus1, Locus2, Overall, n_pops_fail) %>% 
  arrange(Locus1, Locus2)
```

Some of these are not surprising since we already know that the two mitochondrial markers are linked, as well as *MHC* and *Tf_ex*.

Since we are most interested in the linkage relationships among the 7 Islands of Divergence RAD SNPs, let's just look at those.

```{r LD_RAD}
RAD_loci <- grep(pattern = "RAD", x = unique(c(LD$Locus1, LD$Locus2)), value = TRUE)

LD %>% 
  filter(Locus1 %in% RAD_loci & Locus2 %in% RAD_loci) %>% 
  select(Locus1, Locus2, Overall, n_pops_fail) %>% 
  ggplot(aes(x = Locus1, y = Locus2, fill = n_pops_fail)) + 
  geom_tile() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + 
  ggtitle("Number of pops with LD P < 0.05\nIslands of Divergence RAD SNPs")
```

Based on the allele frequency plots generated above and the linkage information that we already knew about from Table S4, it looks like we'd be safe adding *One_RAD18507* and *One_RAD27165* to our **inseason marker panel of 24 SNPs**. Since we are adding two SNPs, we need to kick out two of the old ones currently in *loci22*. Based on Tyler's work for the 2012 version of this baseline (see "../2012/2012 Chignik Baseline.xlsx" tab "87 to 24"), we can conclude that the two "best" SNPs to drop would be *One_U503-170* and *One_ZNF-61*.

## Final marker panel decisions
**Add** the following two RAD SNPs from Wes' Islands of Divergence work

1. *One_RAD18507*
2. *One_RAD27165*

**Drop** the following two "old" loci in *loci22* from the 2012 baseline 

1. *One_U503-170*
2. *One_ZNF-61*

Save this updated markerset as *loci22_RAD*.
```{r loci22_RAD}
loci22 <- dget(file = "../../Mixtures/2018/Objects/loci22.txt")

loci22_RAD <-
  sort(c(setdiff(loci22, c(
    "One_U503-170", "One_ZNF-61"
  )), "One_RAD18507", "One_RAD27165"))

save_objects(objects = "loci22_RAD", path = "../2018/Objects/")
```

# Frequency Plots Loci22RAD

I want the allele plots of interest in one file and thought this would be easier than editing the PDF from earlier. Here, I've removed the GPDH and MHC combined markers to allow for plotting. 
```{r AlleleFreq22}
loci20plot <- loci22_RAD[-c(6,8)] # remove combined GPDH and MHC 

Freq22 <- FreqPop.GCL(sillyvec = sillyvec, loci = loci20plot)

Freq22 <-
  Freq22[, , "Allele 1"] / (Freq22[, , "Allele 1"] + Freq22[, , "Allele 2"])

pdf(
  file = "../2018/freqplots/Chignik22SNPfreqplots_popnames.pdf",
  width = 11,
  height = 8.5,
  family = "Times",
  pointsize = 20
)
for (locus in loci20plot) {
  plot(
    Freq22[, locus],
    main = locus,
    col = PopCol,
    pch = 19,
    ylim = c(0, 1),
    ylab = "Freqency",
    xlab = "Collection"
  )
  text(
    x = Freq22[, locus],
    labels = sillyvec,
    cex = 0.7,
    pos = 1,
    srt = 90,
    offset = 1.5,
    xpd = TRUE
  ) # add population names to file
  lines(supsmu(seq(length(sillyvec)), Freq22[, locus]))
}

rm(locus)

dev.off()
```



# Combining Loci

Since I am just using the final 22 loci I need to combine GPDH and MHC2 into their final forms. 
```{r combining loci}
CombineLoci.GCL(sillyvec = sillyvec ,markerset = c("One_GPDH2","One_GPDH"),update = T)

CombineLoci.GCL(sillyvec = sillyvec ,markerset = c("One_MHC2_251","One_MHC2_190"),update = T)
```


### Pooling ###

```{r}
# All loci
(loci105 <- LocusControl$locusnames)
```


Here, I am running through fishers tests using <b>only</b> the final set of 22 loci. 

```{r AlleleFrequenciesRAD22}
Chignik_Freq <-
  FreqPop.GCL(sillyvec = ChignikCollections, loci = loci22_RAD)

str(Chignik_Freq)
```

1) Temporal Pooling Test
The first step is to pool temporal collections together as they are most likely to be similar.  
```{r TemporalPool, warning=FALSE}
ChignikTemporalPool <-
  list(
    c("SCHIA97E", "SCHIA08"),
    c("SCHIG08", "SCHIG98"),
    c("SCLRK97E", "SCLARK08"),
    c("SHAT97E", "SHAT08E")
  )

#perform fisher test on temporal collections
ChignikFisher22RADTemporalResults <- FishersTest.GCL(freq = Chignik_Freq, loci = loci22_RAD, test = ChignikTemporalPool) # Note warnings here but ignored using warning = FALSE for this chunk. Warning is "'hybrid' is ignored for a 2 x 2 table"

sink(file="../2018/pooling/ChignikFisher22RADTemporalResults.csv")
print(ChignikFisher22RADTemporalResults)
sink()

str(ChignikFisher22RADTemporalResults)

ChignikFisher22RADTemporalResults$OverallResults
```

```{r TemporalHistograms}
hist(
  ChignikFisher22RADTemporalResults$ResultsByLocus$SCHIA97E.SCHIA08[, 2],
  main = "SCHIA97E.SCHIA08",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisher22RADTemporalResults$ResultsByLocus$SCHIA97E.SCHIA08 %>% 
  arrange(pval)

hist(
  ChignikFisher22RADTemporalResults$ResultsByLocus$SCHIG98.SCHIG08[, 2],
  main = "SCHIG98.SCHIG08",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

hist(
  ChignikFisher22RADTemporalResults$ResultsByLocus$SCLRK97E.SCLARK08[, 2],
  main = "SCLRK97E.SCLARK08",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisher22RADTemporalResults$ResultsByLocus$SCLRK97E.SCLARK08 %>% 
  arrange(pval)

hist(
  ChignikFisher22RADTemporalResults$ResultsByLocus$SHAT97E.SHAT08E[, 2],
  main = "SHAT97E.SHAT08E",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisher22RADTemporalResults$ResultsByLocus$SHAT97E.SHAT08E %>% 
  arrange(pval)
```


Due to overll pvalus, locus-specific values, and allele frequency plots I'll only combine "SCHIG98" and "SCHIG08". For all others, the markers with the highest impact are the important ones (i.e., RAD27165, U1004-183).    Further testing on pairs and geographically similar populations will follow. 

-- Pooling --
```{r TemporalPooling}
Groups <- read_csv(file = "../2018/Objects/group_locations.csv") # Import list of silly, groups, and locations

ChignikTemporalPool2 <- list(ChignikTemporalPool[[2]])


lapply(1:length(ChignikTemporalPool2),
       function(x) {
         PoolCollections.GCL(
           collections = ChignikTemporalPool2[[x]],
           loci = loci105,
           IDs = NULL,
           newname = paste(ChignikTemporalPool2[[x]],
                           collapse = ".")
         )
       })

# show just the gcl objects, or samples.
ls(pattern = "^*.*[0-9].*.gcl")

# load temporal pooling from csv
ChignikTemporalCollections <- Groups$ChignikTemporalCollections %>%
  na.omit()

ChignikTemporalFreq <-
  FreqPop.GCL(sillyvec = ChignikTemporalCollections, loci = loci22_RAD)
```


2) Non-temporal pooling tests

For this set, I'll start with pairwise comparisons among early, mid, and late groups then move onto geographic locality. 

```{r Pool1Test, warning=FALSE}
# First, pairwise comparisons for early and late collection. 
ChignikPool1 <-
  list(
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SCHIA97E", "SCHIA97M")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SCHIA08", "SCHIA97M")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SCLRK97E", "SCLARK96")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SCLARK08", "SCLARK96")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT97E", "SHAT96")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT08E", "SHAT96")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT08E", "SHAT97E")]
  )

#perform fisher test on temporal collections
ChignikFisherPool1Results <- FishersTest.GCL(freq = ChignikTemporalFreq, loci = loci22_RAD, test = ChignikPool1)

sink(file="../2018/pooling/ChignikFisherPool1Results.csv")
print(ChignikFisherPool1Results)
sink()

str(ChignikFisherPool1Results)

ChignikFisherPool1Results$OverallResults
```

<i>Overall p-values show that none of the above populations should be pooled. Also, a quic peek at the histograms below confirm this. </i>

```{r Pool1Hist}
hist(
  ChignikFisherPool1Results$ResultsByLocus$SCHIA97E.SCHIA97M[, 2],
  main = "SCHIA97E.SCHIA97M",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

hist(
  ChignikFisherPool1Results$ResultsByLocus$SCHIA08.SCHIA97M[, 2],
  main = "SCHIA08.SCHIA97M",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

hist(
  ChignikFisherPool1Results$ResultsByLocus$SCLRK97E.SCLARK96[, 2],
  main = "SCLRK97E.SCLARK96",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

hist(
  ChignikFisherPool1Results$ResultsByLocus$SCLARK08.SCLARK96[, 2],
  main = "SCLARK08.SCLARK96",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

hist(
  ChignikFisherPool1Results$ResultsByLocus$SHAT97E.SHAT96[, 2],
  main = "SHAT97E.SHAT96",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

hist(
  ChignikFisherPool1Results$ResultsByLocus$SHAT08E.SHAT96[, 2],
  main = "SHAT08E.SHAT96",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)
```

## Pooling tests 2 ##

```{r Pool2Test, warning=FALSE}
# First, pairwise comparisons for early and late for each temporal collection. 
ChignikPool2 <-
  list(
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT97E", "SCUCU08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT96", "SCUCU08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT08E", "SCUCU08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT96", "SCLARK96")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT96", "SCLRK97E")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT96", "SCLARK08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT97E", "SCLARK96")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT97E", "SCLRK97E")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT97E", "SCLARK08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT08E", "SCLARK08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT08E", "SCLARK08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SHAT08E", "SCLARK08")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SFAN97", "SALEC97")],
    ChignikTemporalCollections[ChignikTemporalCollections %in% c("SBROAD97", "SBSPR97")]
  )

# perform fisher test on temporal collections
ChignikFisherPool2Results <- FishersTest.GCL(freq = ChignikTemporalFreq, loci = loci22_RAD, test = ChignikPool2)

sink(file="../2018/pooling/ChignikFisherPool2Results.csv")
print(ChignikFisherPool2Results)
sink()

str(ChignikFisherPool2Results)

ChignikFisherPool2Results$OverallResults
```

```{r Pool2Hist}
hist(
  ChignikFisherPool2Results$ResultsByLocus$SCUCU08.SHAT97E[, 2],
  main = "SHAT97E.SCUCU08",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisherPool2Results$ResultsByLocus$SCUCU08.SHAT97E %>% 
  arrange(pval)

hist(
  ChignikFisherPool2Results$ResultsByLocus$SCUCU08.SHAT08E[, 2],
  main = "SCUCU08.SHAT08E",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisherPool2Results$ResultsByLocus$SCUCU08.SHAT08E %>% 
  arrange(pval)
```

<i> Here I only want to pool SALEC97 + SFAN97. The rest of the comparisons dont make sense. Especially in light of allele frequency plots and P-values.</i> 


```{r Pool3}
Groups <- read_csv(file = "../2018/Objects/group_locations.csv") # Import list of silly, groups, and locations

ChignikPool2 <- list(ChignikPool2[[13]])


lapply(1:length(ChignikPool2),
       function(x) {
         PoolCollections.GCL(
           collections = ChignikPool2[[x]],
           loci = loci105,
           IDs = NULL,
           newname = paste(ChignikPool2[[x]],
                           collapse = ".")
         )
       })

# show just the gcl objects, or samples.
ls(pattern = "^*.*[0-9].*.gcl")

# load  pooling from csv
ChignikPool2Collections <- Groups$ChignikPool2Collections %>%
  na.omit()

ChignikPool2Freq <-
  FreqPop.GCL(sillyvec = ChignikPool2Collections, loci = loci22_RAD)
```


## Pooling tests 3 ##

```{r Pool3Test, warning=FALSE}
# First, pairwise comparisons for early and late for each temporal collection. 
ChignikPool3 <-
  list(
    ChignikPool2Collections[ChignikPool2Collections %in% c("SBOUL97", "SBSPR97")],
    ChignikPool2Collections[ChignikPool2Collections %in% c("SBOUL97", "SBROAD97")],
    ChignikPool2Collections[ChignikPool2Collections %in% c("SBOUL97", "SALEC97.SFAN97")], 
    ChignikPool2Collections[ChignikPool2Collections %in% c("SBROAD97", "SALEC97.SFAN97")],
    ChignikPool2Collections[ChignikPool2Collections %in% c("SBSPR97", "SALEC97.SFAN97")]
  )

# perform fisher test on temporal collections
ChignikFisherPool3Results <- FishersTest.GCL(freq = ChignikPool2Freq, loci = loci22_RAD, test = ChignikPool3)

sink(file="../2018/pooling/ChignikFisherPool3Results.csv")
print(ChignikFisherPool3Results)
sink()

str(ChignikFisherPool3Results)

ChignikFisherPool3Results$OverallResults
```
...noting to see here, move along. 

```{r Pool3Hist}
hist(
  ChignikFisherPool3Results$ResultsByLocus$SBOUL97.SBSPR97[, 2],
  main = "SBOUL97.SBSPR97",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisherPool3Results$ResultsByLocus$SBOUL97.SBSPR97 %>% 
  arrange(pval)

hist(
  ChignikFisherPool3Results$ResultsByLocus$SBOUL97.SBROAD97[, 2],
  main = "SBOUL97.SBROAD97",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisherPool3Results$ResultsByLocus$SBOUL97.SBROAD97 %>% 
  arrange(pval)

hist(
  ChignikFisherPool3Results$ResultsByLocus$SBOUL97.SALEC97.SFAN97[, 2],
  main = "SBOUL97.SALEC97.SFAN97",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisherPool3Results$ResultsByLocus$SBOUL97.SALEC97.SFAN97 %>% 
  arrange(pval)

hist(
  ChignikFisherPool3Results$ResultsByLocus$SBROAD97.SALEC97.SFAN97[, 2],
  main = "SBROAD97.SALEC97.SFAN97",
  breaks = seq(0, 1, by = 0.05),
  xlab = "p-value"
)

ChignikFisherPool3Results$ResultsByLocus$SBROAD97.SALEC97.SFAN97 %>% 
  arrange(pval)
```

<i> None of the tests from Pool3 are great. The 'best' pairs are driven by markers that show clear differences in allele frequencies thus I chose to not pool. </i>    

### Pooling Conclusions ###

After examining temporal collections and conducting pooling tests, I've settled on 16 populations for the Chignik 2018 baseline. This is a change from the previous 7 populations and is driven highly by the new RAD markers. Note that one population (SWESTF97) was excluded from this baseline due to the lack of metadata. The 2012 baseline included this group but the samples were eventually dropped from analyses. The new baseline consists of the following populations: 
    
    SBOUL97, SBROAD97, SBSPR97, SCHIA97E, SCHIA08, SCHIA97M, SCHIG98.SCHIG08, SCLRK97E, SCLARK08, SCUCU08, SHAT97E, SHAT08E, SWESTF08, SHAT96, SCLARK96, SALEC97.SFAN97
    

### Testing baseline ### 

Prior to testing the baseline, I need to save a few objects from above. Specifically, I am interested in the groupvec, populations list, loci, and groups. 

```{r save_objects}
# Save populatrion list, making sure its in a geographic ordering similar to Tylers 2012 work.
Chignik16pops <-
  c(    
    "SBROAD97",
    "SBSPR97",
    "SBOUL97",
    "SALEC97.SFAN97",
    "SCHIA08",
    "SCHIA97E",
    "SCHIA97M",
    "SWESTF08",
    "SCUCU08",
    "SHAT96",
    "SHAT97E",
    "SHAT08E",
    "SCLARK96",
    "SCLRK97E",
    "SCLARK08",
    "SCHIG98.SCHIG08"
  )

Chignik16pops <- dget(file = "../2018/Objects/Chignik16pops.txt")

save_objects(objects = "Chignik16pops", path = "../2018/Objects/")

# create the new groupvecs which onsist of Black Lake or Chignik Lake (1 or 2, respectively)
groupvec16 <- c(1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2)

# save groupvec16 object. Note that this order must allgin with the order of the Chignik16pops above. 
save_objects(objects = "groupvec16", path = "../2018/Objects/")

# load groupvec16 object
groupvec16 <- dget(file = "../2018/Objects/groupvec16.txt")

#Double-check the order - does this look correct!?
Chignik16pops
groupvec16

group_names <- dget(file = "../2018/Objects/ChignikGroups.txt")
```

Now move onto testing! 


```{r Baseline}
BaselineFortran <- CreateBaseline.GCL(sillyvec = Chignik16pops, loci = loci22_RAD, dir = "../2018/BAYES/Baseline", basename = "Chignik16pops22RAD", type = "BAYES", groupvec = groupvec16)

save_objects(objects = "BaselineFortran", path = "../2018/Objects/")
```


### New pops + old markers ### 

```{r newpop_oldmark}

loci24 <- dget(file = "../2012/Objects/loci24.txt")

NewpopOldmarkFortran <- CreateBaseline.GCL(sillyvec = Chignik16pops, loci = loci24, dir = "../2018/BAYES/Baseline", basename = "Chignik16pops24loci", type = "BAYES", groupvec = groupvec16)

save_objects(objects = "NewpopOldmarkFortran", path = "../2018/Objects/")
```

### Rubias testing ###


Now that I have the final baseline picked I can carry out testing via rubias. Here, I am going to test three variations of the baseline:    
1) Old populations, old markers
2) Old populations, new markers
3) New populations, new markers


```{r}
p_load_gh("eriqande/rubias")
```

First, I need all .gcl objects, loci, and pops in my environment so I will create the old gcls. 
```{r PoolingOld}
Groups <- read_csv(file = "../2018/Objects/group_locations.csv") # Import list of silly, groups, and locations


ChignikPoolOld <- list(
  c("SBROAD97", "SBSPR97", "SBOUL97", "SFAN97", "SALEC97"),
  c("SCHIA08", "SCHIA97E", "SCHIA97M"),
  c("SWESTF08"),
  c("SCUCU08", "SHAT08E", "SHAT97E"),
  c("SHAT96", "SCLARK96"),
  c("SCLARK08", "SCLRK97E")
)

lapply(1:length(ChignikPoolOld),
       function(x) {
         PoolCollections.GCL(
           collections = ChignikPoolOld[[x]],
           loci = loci105,
           IDs = NULL,
           newname = paste(ChignikPoolOld[[x]],
                           collapse = ".")
         )
       })

# show just the gcl objects, or samples.
ls(pattern = "^*.*[0-9].*.gcl")

Chignik7pops <- Groups$Chignik7Populations[!is.na(Groups$Chignik7Populations)]

groupvec7 <- dget(file = "../2012/Objects/Groupvec7.txt")
```


Now, we should have evertything to create the rubias baseline. 

```{r CreateBaseline}
oldPop_oldMark <-
  create_rubias_baseline(
    sillyvec = Chignik7pops,
    loci = loci24,
    group_names = group_names,
    groupvec = groupvec7,
    path = "../2018/rubias/",
    baseline_name = "oldPopoldMark"
  )

newPop_oldMark <-
  create_rubias_baseline(
    sillyvec = Chignik16pops,
    loci = loci24,
    group_names = group_names,
    groupvec = groupvec16,
    path = "../2018/rubias/",
    baseline_name = "newPopoldMark"
  )

newPop_newMark <-
  create_rubias_baseline(
    sillyvec = Chignik16pops,
    loci = loci22_RAD,
    group_names = group_names,
    groupvec = groupvec16,
    path = "../2018/rubias/",
    baseline_name = "newPopnewMark"
  )
```


Then assess reference using leave one out
```{r LOO}
# Old populations and old markers
set.seed(12)
loo_OPOM <- assess_reference_loo(
  reference = oldPop_oldMark,
  gen_start_col = 5,
  reps = 50,
  mixsize = 200
)

# New Populations and old Markers
set.seed(12)
loo_NPOM <- assess_reference_loo(
  reference = newPop_oldMark,
  gen_start_col = 5,
  reps = 50,
  mixsize = 200
)

# New populations and new Markers
set.seed(12)
loo_NPNM <- assess_reference_loo(
  reference = newPop_newMark,
  gen_start_col = 5,
  reps = 50,
  mixsize = 200
)
```


Now, we can summarise the output by reporting unit...

```{r SummLOO}
# old populations and old markers
tmpOPOM <- loo_OPOM %>%
  group_by(iter, repunit) %>%
  summarise(
    true_repprop = sum(true_pi),
    reprop_posterior_mean = sum(post_mean_pi),
    repu_n = sum(n)
  ) %>%
  mutate(repu_n_prop = repu_n / sum(repu_n))


# old populations and new markers
tmpNPOM <- loo_NPOM %>%
  group_by(iter, repunit) %>%
  summarise(
    true_repprop = sum(true_pi),
    reprop_posterior_mean = sum(post_mean_pi),
    repu_n = sum(n)
  ) %>%
  mutate(repu_n_prop = repu_n / sum(repu_n))

# new populations and new markers
tmpNPNM <- loo_NPNM %>%
  group_by(iter, repunit) %>%
  summarise(
    true_repprop = sum(true_pi),
    reprop_posterior_mean = sum(post_mean_pi),
    repu_n = sum(n)
  ) %>%
  mutate(repu_n_prop = repu_n / sum(repu_n))
```

Or plot comparing to their "n" value, which is the actual number of fish from each 
reporting unit in the sample.

```{r plot-n}
#old pops and old markers
ggplot(tmpOPOM,
       aes(x = repu_n_prop, y = reprop_posterior_mean, colour = repunit)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  facet_wrap( ~ repunit) +
  ggtitle("Old Populations and Old Markers")

#new pops and old markers
ggplot(tmpNPOM,
       aes(x = repu_n_prop, y = reprop_posterior_mean, colour = repunit)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  facet_wrap( ~ repunit) +
  ggtitle("New Populations and Old Markers")

#new pops and new markers
ggplot(tmpNPNM,
       aes(x = repu_n_prop, y = reprop_posterior_mean, colour = repunit)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) +
  facet_wrap( ~ repunit) +
  ggtitle("New Populations and New Markers")
```


```{r SessionInfo}
# for record keeping
sessionInfo()
```


--- <i> CODE ARCHIVE NOTES </i> ---

Initially, I carried out pooling similar to Tyler's original 2012 baseline and found large differences at pooling test 1. Temportal locations showed some differences in overall p-values but the pooling test 1 lead to significant changes in overall p-values and changes in the individual values. I tested one  pooled collection, "SHAT96", "SCLARK96", and found that our RAD marker of interest was driving these chances. This marker was selected based on LD and Wes Larson's paper as described in the <i> Linkage disequilibrium </i> section above.    

I decided to re-run the pooling steps using ONLY the markers of interest. These are listed in object "loci22_RAD" and are as follows:     
    
     c("One_ACBP-79", "One_agt-132", "One_cin-177", "One_CO1", "One_ghsR-66", "One_GPDH2.One_GPDH", "One_HpaI-436", "One_MHC2_251.One_MHC2_190", "One_ODC1-196", "One_RAD18507", "One_RAD27165", "One_RAG3-93", "One_redd1-414", "One_sys1-230", "One_Tf_ex3-182", "One_U1003-75", "One_U1004-183", "One_U1009-91", "One_U1012-68", "One_U1016-115", "One_U1209-111", "One_U1212-106") \
     
I ended up with 16 populatioons instead of the original 7! 

--- <i> END CODE ARCHIVE </i> ---